### PR TITLE
Fixes#1354 : Add Functionality to Edit Badge Name

### LINKED ISSUE #1354: Edit the Badge Name Feature
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behaviour:**

**Expected behaviour:**

**Steps to reproduce:**
Working on it

### PR DESCRIPTION
**Pull request by ParthS007**
Fixes #1354 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Add Functionality to Edit Badge Name

image

### CODE REVIEW

File: frontend/app/components/user-component/my-badges.js [hunk 1]
```diff
@@ -21,6 +21,32 @@ export default Component.extend({
         });
     },
 
+    updateBadgeName() {
```
Comment (yashLadha) [inline frontend/app/components/user-component/my-badges.js]: Can ask for the `badge` as the parameter and send it to parent controller to update it as it happens in real time in the store.

File: frontend/app/templates/components/user-component/my-badges.hbs [hunk 2]
```diff
@@ -12,7 +12,15 @@
             <img src="{{badge.image_link}}">
           </div>
           <div class="content">
-            <div class="header">{{badge.badge_name}}</div>
+            <div class="ui form" {{action 'updateBadgeName' on="submit"}}>
```
Comment (yashLadha) [inline frontend/app/templates/components/user-component/my-badges.hbs]: send `badge` as the parameter

**Comment by ParthS007** [comment]
@yashLadha  Done with the changes but Badge Name is not changing on submitting the form. Please take a look again.

**Comment by ParthS007** [comment]
@yashLadha  Can you please update your review?

**Comment by gabru-md** [comment]
merging this

**Pull request merged**